// @kode-frontend/react-native-biometry-tools/lib/commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.BiometryType = exports.BiometryErrorCode = void 0;

var _reactNative = require("react-native");

let BiometryType;
exports.BiometryType = BiometryType;

(function (BiometryType) {
  BiometryType["FACE_ID"] = "Face ID";
  BiometryType["TOUCH_ID"] = "Touch ID";
  BiometryType["FINGERPRINT"] = "Fingerprint";
  BiometryType["FACE"] = "Face";
  BiometryType["IRIS"] = "Iris";
})(BiometryType || (exports.BiometryType = BiometryType = {}));

let BiometryErrorCode;
exports.BiometryErrorCode = BiometryErrorCode;

(function (BiometryErrorCode) {
  BiometryErrorCode["NOT_ENROLLED"] = "BiometryScannerNotEnrolled";
  BiometryErrorCode["NOT_SUPPORTED"] = "BiometryScannerNotSupported";
  BiometryErrorCode["NOT_AVAILABLE"] = "BiometryScannerNotAvailable";
  BiometryErrorCode["PASSCODE_NOT_SET"] = "PasscodeNotSet";
  BiometryErrorCode["DEVICE_LOCKED_PERMANENT"] = "DeviceLockedPermanent";
})(BiometryErrorCode || (exports.BiometryErrorCode = BiometryErrorCode = {}));

const {
  BiometryTools
} = _reactNative.NativeModules;
var _default = BiometryTools;
exports.default = _default;
